be/tool/beck.c

```diff
@@ -882,9 +1012,12 @@ static int bnode(struct scanner *s, struct rectype *r, char *buf)
 
 	if (!IS_IN_ARRAY(idx, bt) || bt[idx].b_type == 0)
 		idx = ARRAY_SIZE(bt) - 1;
+
 	genadd(node->bt_backlink.bli_gen);
-	if (s->s_gen != 0 && s->s_gen != node->bt_backlink.bli_gen)
-		return 0;
+	if (!s->s_gen_found) {
+		s->s_gen_found = true;
+		s->s_gen = node->bt_backlink.bli_gen;
+	}
```

Comment:
[Suggestion]This can be made a single function.

Reply:
Yes probably an inline function. We can do that during refactoring exercise next week.